client/my-sites/plugins/controller.js

```diff
@@ -111,7 +115,25 @@ export function browsePluginsOrPlugin( context, next ) {
 }
 
 export function browsePlugins( context, next ) {
-	renderPluginsBrowser( context );
+	const searchTerm = context.query.s;
+	let pluginsResultsComponent = PluginsDiscoveryPage;
+	if ( searchTerm ) {
+		pluginsResultsComponent = PluginsSearchResultPage;
+	}
+	renderPluginsBrowser( context, pluginsResultsComponent );
+	next();
+}
+
+export function browsePluginsCategory( context, next ) {
+	let pluginsResultsComponent = PluginsCategoryResultsPage;
+
+	// Temporary condition until all searches uses /plugin url
```

Comment:
@lupus2k Are there temporary fixes here because we split the PR?

Reply:
I think it chould be not `temporary` but if some others take a look to confirm this, that would be great.

Reply:
Is there a way to get search out of the category page? Similar, getting search and categories out of the discover page. It might help with these unit test failures too as you wouldn't need to pass through the elements to create.

Reply:
We are leaving search in categories as a temporary fix because this manages `/plugins/browse/:category` and the search still uses current URL to search, for example `/plugins/browse/seo?s=ecommerce`.
We can make a redirect in this route like `/plugins/browse/seo?s=ecommerce` => `/plugins?s=ecommerce` or we can make all search goes to `/plugins?s=`